x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/mapper/SemanticTextFieldMapper.java

```diff
@@ -398,6 +400,25 @@ public Object getOriginalValue(Map<String, Object> sourceAsMap) {
         return XContentMapValues.extractValue(TEXT_FIELD, fieldValueMap);
     }
 
+    @Override
+    protected void doValidate(MappingLookup mappers) {
+        int parentPathIndex = fullPath().lastIndexOf(leafName());
+        if (parentPathIndex > 0) {
+            // Check that the parent object field allows subobjects.
+            // Subtract one from the parent path index to omit the trailing dot delimiter.
+            ObjectMapper parentMapper = mappers.objectMappers().get(fullPath().substring(0, parentPathIndex - 1));
+            if (parentMapper == null) {
+                throw new IllegalStateException(CONTENT_TYPE + " field [" + fullPath() + "] does not have a parent object mapper");
+            }
+
+            if (parentMapper.subobjects() == ObjectMapper.Subobjects.DISABLED) {
+                throw new IllegalArgumentException(
+                    CONTENT_TYPE + " field [" + fullPath() + "] cannot be in an object field with subobjects disabled"
+                );
+            }
+        }
+    }
```

Comment:
We got a two-for-one with this bug fix 😁 

Since the semantic text field type adds sub-objects to the mapping upon first document ingestion, it cannot be in an object field with subobjects disabled. Before this fix, doing so resulted in Very Bad Things™, such as the [document parser handling inference sub-fields as runtime fields and triggering assertion failures](https://github.com/Mikep86/elasticsearch/blob/516aa119453d97ec7ac58292e8a91642a5d75893/server/src/main/java/org/elasticsearch/index/mapper/DocumentParser.java#L1039-L1039).